packages/cli-e2e/utils/terminal/do.ts

```diff
@@ -153,6 +161,11 @@ export class Do implements Promise<void> {
       onParams = [condition, () => callback(this.action.process, resolve)];
     }
 
+    if (isConditionPromise(condition)) {
+      condition.then(() => callback(this.action.process, resolve));
+      return;
```

Comment:
A new type of 'Condition': Promises.

They replace the original 'function' promises but are much simpler to work with, especially combined with other terminal toolchains (see what's done in the spec refactor)